client/blocks/reader-feed-item/index.tsx

```diff
@@ -60,7 +60,11 @@ export default function ReaderFeedItem( props: ReaderFeedItemProps ): JSX.Elemen
 
 	// Fetch feed and site data.
 	const queryFeed: boolean = ! isWpcomFeed; // No need to query feed data for WPCOM feeds.
-	const { data: feed, isLoading: isFeedLoading } = Reader.useReadFeedQuery( queryFeed, feedId );
+	const {
+		data: feed,
+		isLoading: isFeedLoading,
+		refetch: refetchFeed, // For cache invalidation and manually refetching the feed data.
+	} = Reader.useReadFeedQuery( queryFeed, feedId );
 	const { data: site, isLoading: isSiteLoading } = Reader.useReadFeedSiteQuery( Number( blogId ) );
```

Comment:
Im confused why are are getting both of these, but our logic below for whether or not the user is subscribed always seems to be based on the `feed` result only, and we only refetch the feed version. 

A lot of old logic around here seems to be written like either a subscription is based on a blogId OR a feedId, and we have separate handling for both. Is that outdated / is everything handled by feedId now?

Is there any case where we are subscribed by blogId and not feedId? If not, is the site version here only being used to get some specific display information (?), should this information be present on the feed result anyways?

Reply:
I have [refactored this code](https://github.com/Automattic/wp-calypso/pull/101011) and also struggled to find a case where we only use siteID to display the information but kept the functionality as is to avoid any surprises. We can definitely look deeper if we want to remove site fetch.

Reply:
> Is there any case where we are subscribed by blogId and not feedId? If not, is the site version here only being used to get some specific display information (?)

Based on quick testing this seems like the cause.

> should this information be present on the feed result anyways?

It should be but I'm not sure if we can gurantee it always like for newly connected Jetpack sites that no one have subscribed yet. There will be other cases as well.